src/CommunityToolkit.Maui/Views/Expander/Expander.shared.cs

```diff
@@ -132,12 +130,42 @@ public ExpandDirection Direction
 
 	Grid ContentGrid => (Grid)base.Content;
 
-	/// <inheritdoc/>
-	protected override async void OnParentChanged()
+	/// <summary>
+	/// Overridable method that executes each time the <see cref="Header"/> is tapped
+	/// </summary>
+	/// <param name="tappedEventArgs">The <see cref="TappedEventArgs"/> provided by <see cref="Expander.tapGestureRecognizer"/>.</param>
+	public virtual void OnHeaderTapped(TappedEventArgs tappedEventArgs)
```

Comment:
@pictos @VladislavAntonyuk 

FYI - I simplified the code from #890 a bit, removing the `public static Action`, replacing it with `public virtual void OnHeaderTapped()`.

I configured Expander to [run `OnHeaderTapped()` every time `Expander.tapGestureRecognizer` is triggered](https://github.com/CommunityToolkit/Maui/pull/751/files#diff-a7933be10a281abb7a9062799e5eb3b5e633518aaff6326c9a8d1b4adc829a6bR251).

This ensures that `Expander` supports `CollectionView` by default, and it allows users to override this method, implementing their own logic if they'd like. It also allows us to one day remove the code inside `OnHeaderTapped()` if/when .NET MAUI fixes the bug on their end without causing any breaking changes to MCT devs.

Reply:
We should also add a Unit Test for this new method since it is `public`, if we can

Reply:
With public virtual method user needs to inherit from Expander and create his own control.
With public static property he only need to create an action and assign it.
Public static property seems to be a simpler solution from user perspective. We can mark that property obsolete and remove it with major release.

Reply:
> With public virtual method user needs to inherit from Expander and create his own control. With public static property he only need to create an action and assign it. Public static property seems to be a simpler solution from user perspective

True, but let's zoom out a bit: How many developers are actually going to change the default behavior? My guess is not many, if any. 

If a developer is going to go through the trouble of changing the default behavior of `Expander`, I doubt they'll mind creating a custom Expander class. It doesn't require many lines of code:

```cs
class CustomExpander : Expander
{
  public virtual void OnHeaderTapped(TappedEventArgs tappedEventArgs)
  {
    base.OnHeaderTapped(tappedEventArgs);
    
    // custom code
  }
}
```

> We can mark that property obsolete and remove it with major release

My goal is to avoid us purposely adding a property that we knowingly will `[Obsolete]` later because `[Obsolete]` code adds technical debt and becomes a headache to manage in the code + the docs.

tl;dr - I prefer that we be selfish and avoid purposely adding technical debt, even if it requires a small bit more work for the developer to create a new subclass to override the default behavior

Reply:
@brminnick, I would recommend keeping the `Action` approach, take a sample of how you can customize the PlatformView on .NET MAUI `EntryHandler.PlatformFactoryView = (handler) => return PlatformView;`, the idea .NET MAUI is to avoid inheritance and have external points of extensibility.

Reply:
> My goal is to avoid us purposely adding a property that we knowingly will [Obsolete] later because [Obsolete] code adds technical debt and becomes a headache to manage in the codebase + the docs.

We don't need to Obsolete that, we can make it available for anyone that want to influence the Expand behavior, we will need to Obsolete the default methods when that be fixed.

Reply:
Let's again zoom out: Why are we even adding a public API? Are devs demanding this? 

If not, let's make `OnHeaderTapped` a private method.

I don't think many, if any, developers are going to modify the default functionality.

Reply:
Sorry, but I still can't see your point be an argument against the usage of an Action instead of `virtual` method.

Reply:
My question is Why are we adding this new API? Are developers asking for this? Do we have an open Issue for this?

I understand it's a workaround to adding Expander support to CollectionView + ListView. But, this PR is created to _fix_ Expander support for CollectionView + ListView, right?

So once this PR is merged, we won't need this new API, right?

Reply:
This PR, IMHO, doesn't **fix** the issue, it provides a workaround since the fix should be in the .NET MAUI. We merged a PR on MAUI to improve the workaround (for .net 8), making it possible to get access to the members that we need without reflection.
So, since this is a workaround devs can opt-in to use ours or create their own implementation, with that we don't have the _obligation_ to support all possible scenarios (I would imagine there's at least one scenario that our workaround will fail) so let's say we provide a fix for 80% of the cases and the other 20% will have an easy way to implement their own fix.

This new API (Action) is a point of extensibility that devs can use to implement their own delegate in other to control that behavior, the inspiration come from [here](https://github.com/dotnet/maui/blob/3bae70081e3bb2639cf558e8351454368ac811c5/src/Core/src/Handlers/View/ViewHandlerOfT.cs#L58), you can see the use at line 72 of the same file. The idea is to have a place that devs can customize without need to extend the control, what's normal on XF but not on .NET MAUI

Reply:
Ah - got it. 

Thanks man - I remember the conversation from our standup now. 

I'll clean up my mess in the morning 👍

Reply:
✅ `public virtual void` reverted to `void`
✅ `Action` added with updated XML documentation